File: 0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set=new HashSet<>();
        int i=0,j=0,max=0;
        while(j<s.length()){
            while(set.contains(s.charAt(j)) && i<j){
                set.remove(s.charAt(i));
                i++;
            }
            set.add(s.charAt(j));
            if(set.size()>max) max=set.size();
            j++;
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        cont=set()
        max_o=0
        l=0
        r=0
        while(r<len(s)):
            if s[r] not in cont:
                cont.add(s[r])
                r+=1
                max_o=max(len(cont),max_o)
            else:
                for i in range(l,r):
                    cont.remove(s[i])
                    if s[i]==s[r]:
                        l=i+1
                        break
        return max_o

